python/ray/rllib/optimizers/apex_optimizer.py

```diff
@@ -110,13 +127,22 @@ def step(self):
             ra, replay = self.inqueue.get()
         if replay is not None:
             with self.grad_timer:
-                td_error = self.local_evaluator.compute_apply(replay)
+                td_error = self.local_evaluator.compute_apply(replay)[
+                    "td_error"]
             self.outqueue.put((ra, replay, td_error))
         self.learner_queue_size.push(self.inqueue.qsize())
         self.weights_updated = True
 
 
-class ApexOptimizer(Optimizer):
+class ApexOptimizer(PolicyOptimizer):
+    """Main event loop of the Ape-X optimizer.
+
+    This class coordinates the data transfers between the learner thread,
+    remote evaluators (Ape-X actors), and replay buffer actors.
+
+    This optimizer requires that policy evaluators return an additional
+    "td_error" array in the info return of compute_gradients(). This error
+    term will be used for sample prioritization."""
 
     def _init(
```

Comment:
One thing that is a little confusing and not very apparent in the docs is where all these parameters are being passed in. Reading this code, one would have to do some digging to jump through the various abstractions (ie, `ApexAgent` -^ `DQNAgent` -> `ApexOptimizer` -v `PolicyOptimizer` -^ `ApexOptimizer` to realize the chain of method calls needed to do this. 

Providing some note in the documentation page, and also a small comment here would be good.

Reply:
After all, this is essentially exposed to user.